### PR TITLE
Comparisons

### PR DESCRIPTION
**Pull request by JoeGab**

### CODE REVIEW

File: 1-js/02-first-steps/08-comparison/article.md [hunk 1]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
 
-## Summary
+## Zusammenfassung
 
-- Comparison operators return a boolean value.
-- Strings are compared letter-by-letter in the "dictionary" order.
-- When values of different types are compared, they get converted to numbers (with the exclusion of a strict equality check).
-- The values `null` and `undefined` equal `==` each other and do not equal any other value.
-- Be careful when using comparisons like `>` or `<` with variables that can occasionally be `null/undefined`. Checking for `null/undefined` separately is a good idea.
+- Vergleichende Operatoren geben einen boolschen Wert zurück.
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
- Vergleichsoperatoren geben einen boolschen Wert zurück.
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 2]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
+- Strings (Zeichenfolgen) werden Buchstabe für Buchstabe in der "Wörterbuch"-Reihenfolge verglichen.
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
- Strings werden Buchstabe für Buchstabe in der "Wörterbuch"-Reihenfolge verglichen.
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 3]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
+- Wenn Werte unterschiedlichen Typs verglichen werden, werden sie in Zahlen (`number`) umgewandelt (außer bei der strikten Gleichheitsprüfung).
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
- Wenn Werte verschiedener Typen verglichen werden, werden sie in Zahlen umgewandelt (unter Ausschluss einer strikten Gleichheitsprüfung).
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 4]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
+Dies geschieht, weil Operanden unterschiedlichen Typs mit dem Gleichheitsoperator `==` in `number` umgewandelt werden. Ein leerer String wird, genau wie bei `false`, zu einer numerischen Null.
 
-What to do if we'd like to differentiate `0` from `false`?
+Was ist zu tun, wenn wir `0` von `false` unterscheiden wollen?
 
-**A strict equality operator `===` checks the equality without type conversion.**
+**Der strikte Gleichheitsoperator `===` überprüft auf Gleichheit ohne Datentypumwandlung.**
 
-In other words, if `a` and `b` are of different types, then `a === b` immediately returns `false` without an attempt to convert them.
+Mit anderen Worten, wenn `a` und `b` von unterschiedlichem Datentyp sind, dann gibt `a === b` sofort `false` zurück, ohne einen Versuch, sie zu konvertieren.
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
Mit anderen Worten, wenn `a` und `b` unterschiedliche Typen sind, gibt `a === b` sofort `false` zurück, ohne sie zu konvertieren.
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 5]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
+Es gibt auch einen "strikten Ungleichheits"-Operator `!==` analog zu `!=`.
 
-The strict equality operator is a bit longer to write, but makes it obvious what's going on and leaves less room for errors.
+Der Operator für strikte Gleichheit ist etwas länger zu schreiben, macht aber deutlich, was vor sich geht, und lässt weniger Raum für Fehler.
 
-## Comparison with null and undefined
+## Vergleiche mit null und undefined
 
-There's a non-intuitive behavior when `null` or `undefined` are compared to other values.
+Es gibt ein nicht-intuitives Verhalten, wenn `null` oder `undefined` mit anderen Werten verglichen wird.
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
Es gibt ein nicht intuitives Verhalten, wenn `null` oder `undefined` mit anderen Werten verglichen wird.
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 6]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
+Für die strikte Gleichheitsprüfung `===`
+: sind diese Werte unterschiedlich, weil jeder von ihnen ein verschiedener Datentyp ist.
 
     ```js run
     alert( null === undefined ); // false
     ```
 
-For a non-strict check `==`
-: There's a special rule. These two are a "sweet couple": they equal each other (in the sense of `==`), but not any other value.
+Für die nicht-strikte Gleichheitsprüfung `==`
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
Für die nicht strikte Kontrolle `==`
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 7]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
+: gibt es eine besondere Regel. Diese beiden sind ein "süßes Pärchen": sie sind gleich (im Sinne von `==`), aber nicht irgendeinem anderen Wert.
 
     ```js run
     alert( null == undefined ); // true
     ```
 
-For maths and other comparisons `< > <= >=`
-: `null/undefined` are converted to numbers: `null` becomes `0`, while `undefined` becomes `NaN`.
+Für mathematische und andere Vergleiche `< > <= >=`
+: werden `null/undefined` umgewandelt in number (Zahlen): `null` wird zu `0`, während `undefined` zu `NaN` wird.
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
: `null/undefined` werden in Zahlen umgewandelt: `null` wird zu `0`, während `undefined` zu `NaN` wird.
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 8]
```diff
@@ -1,95 +1,95 @@
...
-When comparing values of different types, JavaScript converts the values to numbers.
+Beim Vergleich von Werten verschiedener Datentypen wandelt JavaScript die Werte in `number` (Zahlen) um.
 
-For example:
+Zum Beispiel:
 
 ```js run
-alert( '2' > 1 ); // true, string '2' becomes a number 2
-alert( '01' == 1 ); // true, string '01' becomes a number 1
+alert( '2' > 1 ); // true, der String '2' wird zu number 2
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
alert( '2' > 1 ); // true, String '2' wird zur Zahl 2
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 9]
```diff
@@ -1,95 +1,95 @@
...
+alert( '01' == 1 ); // true, der String '01' wird zu number 1
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
alert( '01' == 1 ); // true, String '01' wird zur Zahl 1
```

File: 1-js/02-first-steps/08-comparison/article.md [hunk 10]
```diff
@@ -101,109 +101,109 @@ alert( Boolean(b) ); // true
...
 alert( null > 0 );  // (1) false
 alert( null == 0 ); // (2) false
 alert( null >= 0 ); // (3) *!*true*/!*
 ```
 
-Mathematically, that's strange. The last result states that "`null` is greater than or equal to zero", so in one of the comparisons above it must be `true`, but they are both false.
+Mathematisch gesehen ist das seltsam. Das letzte Ergebnis besagt, dass "`null` größer oder gleich `0` ist", so dass es in einem der obigen Vergleiche `true` sein muss, aber beide sind `false`.
 
-The reason is that an equality check `==` and comparisons `> < >= <=` work differently. Comparisons convert `null` to a number, treating it as `0`. That's why (3) `null >= 0` is true and (1) `null > 0` is false.
+Der Grund ist, dass die Gleichheitsprüfung `==` und vergleichende Operatoren `> < >= <=` verschieden funktionieren. Vergleichende Operatoren wandeln `null` in eine `number` um, behandeln es als `0`. Deshalb ist (3) `null >= 0` `true` und (1) `null > 0` `false`.
```
Comment (christianhegedues) [inline 1-js/02-first-steps/08-comparison/article.md]:
```suggestion
Der Grund ist, dass die Gleichheitsprüfung `==` und vergleichende Operatoren `> < >= <=` verschieden funktionieren. Vergleichende Operatoren wandeln `null` in eine Zahl um, behandeln es als `0`. Deshalb ist (3) `null >= 0` `true` und (1) `null > 0` `false`.
```

**Review by christianhegedues — CHANGES_REQUESTED** [review]
Good work! I've left a few comments that should be addressed.
